src/comlink.ts

```diff
@@ -475,20 +478,28 @@ function createProxy<T>(
   target: object = function () {}
 ): Remote<T> {
   let isProxyReleased = false;
-  const propProxyCache : Map<(string | symbol), Remote<unknown>> = new Map();
+  const propProxyCache: Map<string | symbol, Remote<unknown>> = new Map();
   const proxy = new Proxy(target, {
     get(_target, prop) {
       throwIfProxyReleased(isProxyReleased);
       if (prop === releaseProxy) {
         return () => {
-          for (const subProxy of propProxyCache.values()) {
-            subProxy[releaseProxy]();
-          }
-          propProxyCache.clear();
           unregisterProxy(proxy);
-          releaseEndpoint(ep);
-          pendingListeners.clear();
           isProxyReleased = true;
+
+          return releaseEndpoint(ep, pendingListeners).finally(() => {
```

Comment:
Do we have to wait for `releaseEndpoint` to resolve before rejecting these promises?